### PR TITLE
[#336] Select which tests to run via command line arguments

### LINKED ISSUE #336: Select which test to run via command line arguments
Currently there is no way to select which test to run:

```haskell
main :: IO ()
main = do
  opts <- execParser optsParser
  let scenario = optsTestScenario opts
  runTests
    [ Test.Cardano.Chain.Block.Bi.tests
    , Test.Cardano.Chain.Block.Validation.tests scenario
    , Test.Cardano.Chain.Common.Lovelace.tests
    , Test.Cardano.Chain.Delegation.Bi.tests
    , Test.Cardano.Chain.Epoch.File.tests
    , Test.Cardano.Chain.Genesis.Json.tests
    , Test.Cardano.Chain.Elaboration.Delegation.tests
    , Test.Cardano.Chain.Ssc.Bi.tests
    , Test.Cardano.Chain.Txp.Bi.tests
    , Test.Cardano.Chain.Txp.Json.tests
    , Test.Cardano.Chain.Txp.Validation.tests scenario
    , Test.Cardano.Chain.Update.Bi.tests
    , Test.Cardano.Chain.Update.Json.tests
    , Test.Cardano.Chain.Block.Validation.Spec.tests
    ]
```

It would be nice if we could select the test to run using `stack` in combination with `file-watch` and `ta` as follows:

```sh
stack test --file-watch --ta "-m *somePattern*"
```

There are options such as using `ghcid`, `ghci` or a combination of both however they have several shortcomings:

- No recompilation when `stack.yaml` or `*.cabal` files are changed
- No recompilation when local dependencies files are changed
- Colored output breaks in some cases

Also it might be nice to add a test framework, to avoid rolling our own.

### PR DESCRIPTION
**Pull request by ruhatch**
Closes #336

### CODE REVIEW

File: test/test.hs [hunk 1]
```diff
@@ -36,33 +40,38 @@ import qualified Test.Cardano.Chain.Update.Properties
...
-    , Test.Cardano.Chain.Update.Properties.tests scenario
-    -- , Test.Cardano.Chain.Block.Validation.Spec.tests scenario
-    ]
+main =
+  mainWithTestScenario
+    $   testGroup "Cardano Ledger Tests"
+    $   tsGroupToTree
+    <$> [ Test.Cardano.Chain.Block.CBOR.tests
+        , Test.Cardano.Chain.Block.Validation.tests
+        -- , Test.Cardano.Chain.Block.Validation.Spec.tests
```
Comment (dnadales) [inline test/test.hs]: Do we need to remove this comment?

Comment (ruhatch) [reply test/test.hs]: Might leave it for now, as we might have to resurrect this stuff

**Comment by dnadales** [comment]
NIce :tada: 

Should we add instructions in the README showing how to select the tests to run?

File: test/test.hs (hunk 1 above)
Comment (dnadales) [reply test/test.hs]: But aren't we running these tests anymore :thinking:

Comment (ruhatch) [reply test/test.hs]: I guess I can delete it for now then!

Comment (dnadales) [reply test/test.hs]: I'm fine with leaving this there. I just thought you forgot to uncomment that test :D

**Review by dnadales — APPROVED** [review]

**Comment by mhuesch** [comment]
Looks like I didn't finish my review before the weekend...

@ruhatch Thanks for doing this, it'll make my test workflow much smoother. Also, I like what you did with `TSGroup` and `concatTSGroups`.

**Pull request merged**